crates/astria-proto/src/native/sequencer/ics20_withdrawal.rs

```diff
@@ -0,0 +1,234 @@
+    #[must_use]
+    pub fn to_fungible_token_packet_data(&self) -> FungibleTokenPacketData {
+        FungibleTokenPacketData {
+            amount: self.amount.to_string(),
+            denom: self.denom.to_string(),
+            sender: self.return_address.to_string(),
+            receiver: self.destination_chain_address.clone(),
+        }
+    }
+
+    #[must_use]
+    pub fn to_raw(&self) -> raw::Ics20Withdrawal {
+        raw::Ics20Withdrawal {
+            amount: Some(self.amount.into()),
+            denom: self.denom.to_string(),
+            destination_chain_address: self.destination_chain_address.clone(),
+            return_address: self.return_address.0.to_vec(),
+            timeout_height: Some(self.timeout_height.into()),
+            timeout_time: self.timeout_time,
+            source_channel: self.source_channel.to_string(),
+        }
+    }
+
+    #[must_use]
+    pub fn into_raw(self) -> raw::Ics20Withdrawal {
+        raw::Ics20Withdrawal {
+            amount: Some(self.amount.into()),
+            denom: self.denom.to_string(),
+            destination_chain_address: self.destination_chain_address,
+            return_address: self.return_address.0.to_vec(),
+            timeout_height: Some(self.timeout_height.into()),
+            timeout_time: self.timeout_time,
+            source_channel: self.source_channel.to_string(),
+        }
+    }
+
+    /// Convert from a raw, unchecked protobuf [`raw::Ics20Withdrawal`].
+    ///
+    /// # Errors
+    ///
+    /// - if the `amount` field is missing
+    /// - if the `denom` field is invalid
+    /// - if the `return_address` field is invalid
+    /// - if the `timeout_height` field is missing
+    /// - if the `source_channel` field is invalid
+    pub fn try_from_raw(proto: raw::Ics20Withdrawal) -> Result<Self, Ics20WithdrawalError> {
+        let amount: Uint128 = proto.amount.ok_or(Ics20WithdrawalError::MissingAmount)?;
+        let return_address = Address::try_from_slice(&proto.return_address)
+            .map_err(Ics20WithdrawalError::InvalidReturnAddress)?;
+        let timeout_height = proto
+            .timeout_height
+            .ok_or(Ics20WithdrawalError::MissingTimeoutHeight)?
+            .into();
+
+        Ok(Self {
+            amount: amount.into(),
+            denom: IbcAsset::from_str(&proto.denom).map_err(Ics20WithdrawalError::InvalidDenom)?,
+            destination_chain_address: proto.destination_chain_address,
+            return_address,
+            timeout_height,
+            timeout_time: proto.timeout_time,
+            source_channel: ChannelId::from_str(&proto.source_channel)
+                .map_err(Ics20WithdrawalError::InvalidSourceChannel)?,
+        })
+    }
+}
+
+impl From<raw::IbcHeight> for IbcHeight {
+    fn from(h: raw::IbcHeight) -> Self {
+        Self {
+            revision_number: h.revision_number,
+            revision_height: h.revision_height,
+        }
+    }
+}
+
+impl crate::native::Protobuf for IbcHeight {
```

Comment:
Since all the fields of `raw::IbcHeight` are `Copy` you don't have to impl all the trait methods. The trait definition provides default implementations for `try_from_raw` (in terms of `try_from_raw_ref`) and `into_raw` (in terms of `to_raw)`. So if you impl `to_raw` and `try_from_raw_ref` you get the other two for free.

`try_from_raw` and `into_raw` can be implemented by non-copy types to avoid allocations.